flask/app.py

```diff
@@ -1991,6 +1991,9 @@ def wsgi_app(self, environ, start_response):
             except Exception as e:
                 error = e
                 response = self.handle_exception(e)
+            except:
```

Comment:
In python2 it's possible to raise exceptions that don't extend even `BaseException`, using `except` guarantees everything will be caught.